In [13]:
import pandas as pd
import re

In [2]:
df1 = pd.read_csv("Xtern_housing.csv")

In [3]:
df2 = pd.read_csv("events.csv")

In [4]:
df3 = pd.read_csv("restaurants.csv")

In [5]:
df1

Unnamed: 0         car        bike
0       The Speak Easy    7.4 miles   7.7 miles
1                zWORKS  18.1 miles  16.1 miles
2        Launch Fishers  22.4 miles  21.9 miles
3  Industrious Mass Ave   2.7 miles   1.8 miles
4          Launch Indy   2.4 miles     15 miles

In [70]:
"""Finding the average distance to travel by bike 
or car to each coworking space"""
ndf1car = []
for car in df1.car:
    car_in_mile = re.sub(' miles',  '', car)    
    ndf1car.append(car_in_mile)

ndf1bike = []
for bike in df1.bike:
    bike_in_mile = re.sub(' miles',  '', bike)    
    ndf1bike.append(bike_in_mile)
    
diff = []
for i in range(0, len(ndf1car)):
    diff.append(str(round(((float(ndf1car[i])+float(ndf1bike[i]))/2), 2)))
df1['average distance'] = diff
df1

Unnamed: 0         car        bike average distance
0       The Speak Easy    7.4 miles   7.7 miles             7.55
1                zWORKS  18.1 miles  16.1 miles             17.1
2        Launch Fishers  22.4 miles  21.9 miles            22.15
3  Industrious Mass Ave   2.7 miles   1.8 miles             2.25
4          Launch Indy   2.4 miles     15 miles              8.7

From the addition of the average distance column, it is clear that the Speak Easy coworkering space is the closest to the housing provided by transportation of bike or car. The average distance of travel from the housing to The Speak Easy coworkering location is 7.55 miles. The next nearest location is the Launch Indy which is 8.7 miles from housing.

In [65]:
df2.head()

event_name type_of_event  \
0          Telly Leung with the Telly Leung Quartet          Music   
1              Buffalo Wabs & The Price Hill Hustle          Music   
2                     Miami Heat vs. Indiana Pacers        Sports    
3  The Tarkington at the Center for the Performin...           Art   
4  Dia de Muertos (Day of the Dead) Community Cel...           Art   

                                             address     date   \
0  924 N. Pennsylvania St., Ste. B, Indianapolis,...  10/23/21   
1  1043 Virginia Ave., Ste. #4, Indianapolis, IN ...  10/22/21   
2    125 S. Pennsylvania St.\nIndianapolis, IN 46204  10/23/21   
3                   3 Center Green\nCarmel, IN 46032  10/29/21   
4      500 W. Washington St., Indianapolis, IN 46204  10/30/21   

   The_Speak_Easy   zWORKS   Launch_Fishers   Industrious_Mass_Ave   \
0              5.4     17.3             16.1                    1.3   
1              7.2     19.6             18.0                    1.4   
2              6.5     18.4             17.2                    0.2   
3             11.2      8.4              8.3                   23.4   
4              6.9     17.9             22.2                    1.0   

   Launch_Indy   
0           1.9  
1           1.6  
2           0.5  
3          16.7  
4           1.1

In [63]:
c1 = df2['The_Speak_Easy '].mean()
c2 = df2['zWORKS '].mean()
c3 = df2['Launch_Fishers '].mean()
c4 = df2['Industrious_Mass_Ave '].mean()
c5 = df2['Launch_Indy '].mean()
print('The_Speak_Easy: ', round(c1, 2), "miles")
print('zWORKS: ', round(c2, 2), "miles")
print('Launch_Fishers: ', round(c3, 2), "miles")
print('Industrious_Mass_Ave: ', round(c4, 2), "miles")
print('Launch_Indy: ', round(c5, 2), "miles")

The_Speak_Easy:  6.47 miles
zWORKS:  14.38 miles
Launch_Fishers:  14.94 miles
Industrious_Mass_Ave:  7.89 miles
Launch_Indy:  6.75 miles


From the second dataframe it is clear that The Speak Easy coworkering location is a hub of many events that interns can participate on a weekly basis. The average distance of travel for nearby events for The Speak Easy is 6.47 miles while Launch_Indy closely follows at 6.75 miles. 

In [66]:
df3.head()

restaurants   \
0                              Nesso   
1                    Greeks Islands    
2                   Yolk - City Way    
3                              Nada    
4  Fogo de Chão Brazilian Steakhouse   

                                       address     cuisine  The_Speak_Easy   \
0      339 Delaware St, Indianapolis, IN 46204     Italian              6.7   
1    906 S Meridian St, Indianapolis, IN 46225       Greek              7.2   
2       220 E South St, Indianapolis, IN 46204      Brunch              6.6   
3     11 W Maryland St, Indianapolis, IN 46204    Mexican               6.4   
4  117 E Washington St, Indianapolis, IN 46204  Brazilian               6.4   

   zWORKS   Launch_Fishers   Industrious_Mass_Ave   Launch_Indy   
0     18.7             21.9                    0.5           0.4  
1     19.0             22.9                    1.0           0.3  
2     18.8             21.8                    0.5           0.4  
3     18.4             21.5                    0.3           0.6  
4     18.6             21.5                    0.3           0.8

In [67]:
c1 = df3['The_Speak_Easy '].mean()
c2 = df3['zWORKS '].mean()
c3 = df3['Launch_Fishers '].mean()
c4 = df3['Industrious_Mass_Ave '].mean()
c5 = df3['Launch_Indy '].mean()
print('The_Speak_Easy: ', round(c1, 2), "miles")
print('zWORKS: ', round(c2, 2), "miles")
print('Launch_Fishers: ', round(c3, 2), "miles")
print('Industrious_Mass_Ave: ', round(c4, 2), "miles")
print('Launch_Indy: ', round(c5, 2), "miles")

The_Speak_Easy:  6.07 miles
zWORKS:  17.78 miles
Launch_Fishers:  20.94 miles
Industrious_Mass_Ave:  1.33 miles
Launch_Indy:  1.42 miles


After calculating, the average distances from the 